File: q40-q49/q48/q48.py
import copy
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        backup = copy.deepcopy(matrix)
        size = len(matrix)
        for i in range(0, size):
            for j in range(0, size):
                matrix[i][j] = backup[size - j - 1][i]
        

File: Sequence/q1-q99/q40-q49/q48/q48.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int size = matrix.size();
        vector<vector<int>> copy(size,vector<int> (size));
        for(int i = 0; i < size; i++) {
            for(int j = 0; j < size; j++) {
                copy[i][j] = matrix[size - 1 - j][i];
            }
        }
        matrix = copy;
    }
};
